py3status/modules/emerge_status.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""
+Emerge_status provides a short information on the current
+emerge status if there is an emerge process currently running
+
+Configuration parameters:
+    cache_timeout: how often we refresh this module in second (default 0)
+    format: *(default '[[\\?if=!hide_if_stopped {prefix}]|'
+                     '[\\?if=is_running {prefix}]]'
+                     '[[\\?if=is_running [\\?if=!total=0 {current}/{total}'
+                     '[\\?if=show_action {action} ]'
+                     '[\\?if=show_pkg {category}/{pkg}]]]|'
+                     '[\\?if=!hide_if_stopped 0/0]]')*
+    hide_if_stopped: Hide all information if no emerge is running (default False)
+    prefix: prefix in statusbar (default "EMRG: ")
+    show_action: En/Disable showing action (default True)
+    show_pkg: Dis-/Enable showing category and pkg name (default True)
+
+Format placeholders:
+    {current} Number of package that is currently emerged
+    {total} Total number of packages that will be emerged
+    {category} Category of the currently emerged package
+    {pkg} Name of the currently emerged packaged
+    {action} Current emerge action
+
+Examples:
+```
+emerge_status {
+    hide_if_stopped = True
+    prefix = emerge
+    show_action = False
+    show_pkg = False
+}
+```
+
+
+@author AnwariasEu
+"""
+
+import re
+import copy
+
+STRING_NOT_INSTALLED = 'not installed'
+
+
+class Py3status:
+    """
+    """
+    cache_timeout = 0
+    format = (
+        # show prefix if either show_stopped or if emerge is running
+        '[[\?if=!hide_if_stopped {prefix}]|[\?if=is_running {prefix}]]'
+        # if emerge is running and total != 0 show
+        '[[\?if=is_running [\?if=!total=0 {current}/{total}'
+        # show action and cat/pkg only if user wants it
+        '[\?if=show_action {action} ][\?if=show_pkg {category}/{pkg}]]]|'
+        # if no emerge is running but show_stopped is set show 0/0
+        '[\?if=!hide_if_stopped 0/0]]')
+    hide_if_stopped = False
+    prefix = "EMRG: "
+    show_action = True
+    show_pkg = True
+
+    def __init__(self):
+        self.ret_default = {
+            'current': 0,
+            'total': 0,
+            'category': "",
+            'pkg': "",
+            'action': "",
+            'is_running': False}
+
+    def _emergeRunning(self):
+        """
+        Check if emerge is running.
+        Returns true if at least one instance of emerge is running.
+        """
+        try:
+            self.py3.command_output(['pgrep', 'emerge'])
+            return True
+        except Exception:
+            return False
+
+    def post_config_hook(self):
+        if not self.py3.check_commands('emerge'):
+            raise Exception("emerge not installed")
+
+    def _getProgress(self):
+        """
```

Comment:
Python naming conventions for functions [[pep link]](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names), ie `_get_progress`. :+1: